sensor/common/processsignal/pipeline.go

```diff
@@ -72,20 +110,28 @@ func (p *Pipeline) Process(signal *storage.ProcessSignal) {
 }
 
 func (p *Pipeline) sendIndicatorEvent() {
-	for indicator := range p.enrichedIndicators {
+	for indicator := range p.cm.GetOutput() {
 		if !p.processFilter.Add(indicator) {
 			continue
 		}
 		p.detector.ProcessIndicator(indicator)
-
-		p.indicators <- message.New(&central.MsgFromSensor{Msg: &central.MsgFromSensor_Event{Event: &central.SensorEvent{
-			Id:     indicator.GetId(),
-			Action: central.ResourceAction_CREATE_RESOURCE,
-			Resource: &central.SensorEvent_ProcessIndicator{
-				ProcessIndicator: indicator,
+		msg := message.New(&central.MsgFromSensor{
+			Msg: &central.MsgFromSensor_Event{
+				Event: &central.SensorEvent{
+					Id:     indicator.GetId(),
+					Action: central.ResourceAction_CREATE_RESOURCE,
+					Resource: &central.SensorEvent_ProcessIndicator{
+						ProcessIndicator: indicator,
+					},
+				},
 			},
-		},
-		},
 		})
+		if !p.centralReady.IsDone() {
```

Comment:
The problem with this approach is that if the pipeline gets disconnected as soon as the message was _already_ sent, there is no way to cancel it, because it was created with `context.Background()` (since you used `message.New`). 

The pattern for the other components is to have a context in the component that is append to every message, and just cancel this context once you go into offline mode.

Reply:
This should be resolved now - see [2b30133](https://github.com/stackrox/stackrox/pull/7011/commits/2b3013309d55021194ebc4c53eb9189c95787c00)